src/app/games/p4/tests/P4Minimax.spec.ts

```diff
@@ -7,16 +7,38 @@ import { P4Rules } from '../P4Rules';
 describe('P4Minimax', () => {
 
     let minimax: P4Minimax;
+    let rules: P4Rules;
 
     beforeEach(() => {
-        const rules: P4Rules = new P4Rules(P4State);
+        rules = new P4Rules(P4State);
         minimax = new P4Minimax(rules, 'P4Minimax');
     });
-    xit('First choice should be center at all IA depths', () => {
+    it('First choice should be center at all IA depths', () => {
         const initialState: P4State = P4State.getInitialState();
         for (let depth: number = 1; depth < 6; depth ++) {
             const node: MGPNode<P4Rules, P4Move, P4State> = new MGPNode(null, null, initialState);
             expect(node.findBestMove(depth, minimax)).toEqual(P4Move.THREE);
         }
     });
+    it('Minimax should prune when instructed to do so', () => {
+        const getBoardValueSpy: jasmine.Spy = spyOn(minimax, 'getBoardValue').and.callThrough();
+        const getListMovesSpy: jasmine.Spy = spyOn(minimax, 'getListMoves').and.callThrough();
+
+        // Given the number of moves of a minimax without alpha-beta pruning
+        rules.node.findBestMove(3, minimax, false, false);
+        const callsToGetBoardValueWithoutPruning: number = getBoardValueSpy.calls.count();
+        getBoardValueSpy.calls.reset();
+        const callsToGetListMovesWithoutPruning: number = getListMovesSpy.calls.count();
+        getListMovesSpy.calls.reset();
+
+        // when computing the same information with alpha-beta pruning enabled
+        rules.node = new MGPNode(null, null, P4State.getInitialState());
+        rules.node.findBestMove(3, minimax, false, true);
+        const callsToGetBoardValueWithPruning: number = getBoardValueSpy.calls.count();
+        const callsToGetListMovesWithPruning: number = getListMovesSpy.calls.count();
+
+        // then the number of calls is strictly lower
+        expect(callsToGetBoardValueWithPruning).toBeLessThan(callsToGetBoardValueWithoutPruning);
+        expect(callsToGetListMovesWithPruning).toBeLessThan(callsToGetListMovesWithoutPruning);
```

Comment:
Wouldn't that be more speaking with numbers ?

Reply:
hm avoir un `toBeLessThan(42)` avec 42 un nombre magique ? (alors que justement le test c'est "faire moins de calcul, peu importe le nombre réel de calcul")

Reply:
Mouais c'est pour ça qu'à l'avenir un bon MinimaxTesting prunable sera plus classe, un qui représente vraiment des arbres, pas ce vieux simulacre ou N.left.right == N.right.left et ou l'arbre est losange :D (aka: un rectangular board :D )
Disons qu'au prochain burnoutte, je fais le TreeDecisionDemo (name tbd) :D